### PR TITLE
Update requirements checking and add a test

### PR DESCRIPTION
**Pull request by LKS90**
Tests are still passing, but nothing ported yet.

ultimate_cron_job_load_all() and _ultimate_cron_get_class('job')

need a new version / equivalent.

### CODE REVIEW

**Comment by LKS90** [comment]
Passes, travis failed with one build.

File: src/Plugin/ultimate_cron/Scheduler/Crontab.php [hunk 1]
```diff
@@ -159,14 +159,15 @@ public function isBehind(CronJob $job) {
     // Check the registered time, and use that if it's available.
     $job_last_ran = $log_entry->start_time;
     if (!$job_last_ran) {
-      $registered = variable_get('ultimate_cron_hooks_registered', array());
-      if (empty($registered[$job->id()])) {
+      $registered = \Drupal::configFactory()->getEditable('ultimate_cron_hooks_registered');
```
Comment (Berdir) [inline src/Plugin/ultimate_cron/Scheduler/Crontab.php]: I have no idea what this was before, but there is no such configuration in 8.x-1.x.

File: ultimate_cron.install [hunk 2]
```diff
@@ -209,7 +207,7 @@ function ultimate_cron_requirements($phase) {
         );
         $requirements['description'][] = t(
           'Some jobs are behind their schedule. Please check if !link is running properly.', array(
-          '!link' => l(t('cron'), 'admin/config/system/cron')
+          '!link' =>  'admin/config/system/cron',
```
Comment (Berdir) [inline ultimate_cron.install]: All links need to use Url::fromRoute().

Comment (Berdir) [inline ultimate_cron.install]: Or Drupal::l(). But it's quite common for transltable strings to include the <a> tags and just inject the url.

**Comment by Berdir** [comment]
Also, missing tests. There's an issue for past that adds a requirements test. This can do something similar.

**Comment by LKS90** [comment]
Regarding the requirements test:

https://www.drupal.org/node/2511822

File: ultimate_cron.install [hunk 3]
```diff
@@ -209,7 +208,7 @@ function ultimate_cron_requirements($phase) {
         );
         $requirements['description'][] = t(
           'Some jobs are behind their schedule. Please check if !link is running properly.', array(
-          '!link' => l(t('cron'), 'admin/config/system/cron')
+          '!link' => \Drupal::l('Cron', Url::fromRoute('system.cron')),
```
Comment (LKS90) [inline ultimate_cron.install]: readd t('Cron')?

Comment (Berdir) [inline ultimate_cron.install]: Yes.

**Comment by LKS90** [comment]
Alright, I moved the test to a WebTestBase test, makes things easier. The route rebuilidng did help the last time, but then I just had more problems because the cron_key was an empty string (I tried resetting the state cache, didn't help). After moving the test to CronJobInstall (which makes sense, considering the requirements are tested in ultimate_cron.install) the route rebuilding is not necessary and I have a cron key.

I also had to fix the isBehind function, which was looking for $job->disabled which doesn't exist anymore, changed it to $job->status().

**Comment by Berdir** [comment]
Nice, merged.

**Pull request merged**